pages/lessons/[slug].js

```diff
@@ -32,9 +54,32 @@ const Transcript = ({url}) => {
   return data ? <Markdown className="prose">{data.text}</Markdown> : null
 }
 
-export default function Lesson({lesson}) {
+const lessonLoader = (slug, token) => (query) => {
+  const authorizationHeader = token && {
+    authorization: `Bearer ${token}`,
+  }
+  const variables = {
+    slug: slug,
+  }
+  const graphQLClient = new GraphQLClient(API_ENDPOINT, {
+    headers: {
+      ...authorizationHeader,
+    },
+  })
+  return graphQLClient.request(query, variables)
+}
+
+export default function Lesson({initialLesson}) {
   const router = useRouter()
   const playerRef = React.useRef(null)
+  const {authToken} = useEggheadUser()
```

Comment:
when and where hooks get called is confusing and a silent killer

Reply:
My understanding: Hooks are always called, but most React hooks (useState, etc) are memo'ed/cached.

So when creating custom hooks, if you're using `useState`, those values should be cached even if it looks like your hook is constantly invoked. I think of hooks like "getters" in that sense.